### DISCUSSION TITLE
How to install through Dockerfile into a Microsoft's Mariner (Azure) Linux deployment

### DISCUSSION THREAD

**Discussion by alerine1** [mono/SkiaSharp#2003] [Q&A]
On Ubuntu this will take care of the dependencies generating a .webP file

FROM mcr.microsoft.com/dotnet/aspnet:6.0-focal AS runtime
RUN apt-get update
RUN apt-get install -y libicu-dev fontconfig-config
ENV DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=true
RUN apt-get install -y libgdiplus linux-libc-dev
COPY . .
ENTRYPOINT ["dotnet", "Yourapp.dll"]

But now I would like to use Mariner 1.x or 2.x and apt-get doesn't work, only way is tdnf <= but packages are not found.

Without the RUN commands on the Docker file regardless if I use the SkiaSharp with or without dependencies, I got an error of a missing dependency. But this only works on Ubuntu apparently, not on Mariner/Cosmic (deployment)

Anyone has been able to generate webp files on Mariner's Linux? Thanks

**Comment by mattleibow** [comment]
Do you know what are the missing dependencies?

Is there a tool like `ldd` or similar on Mariner?

**Reply by alerine1** [reply]
I made SkiaSharp works on Mariner with this:

FROM mcr.microsoft.com/dotnet/aspnet:6.0-cbl-mariner1.0 AS runtime
RUN tdnf check-update
RUN tdnf install icu-devel fontconfig -y
COPY . .
ENTRYPOINT ["dotnet", "YourProject.dll"]
